artifactory.py

```diff
@@ -1031,10 +988,25 @@ def deploy(
         parameters=None,
         explode_archive=None,
         explode_archive_atomic=None,
+        checksum=None,
+        by_checksum=None,
     ):
         """
         Uploads a given file-like object
         HTTP chunked encoding will be attempted
+
+        If by_checksum is True, fobj should be None
```

Comment:
this should be done in code

Reply:
Will raise error if both `by_checksum` and `fobj` are not None.
Thank you for pointing this issue